sql-plugin/src/main/301db/scala/com/nvidia/spark/rapids/shims/v2/GpuRegExpReplaceExec.scala

```diff
@@ -48,6 +50,14 @@ class GpuRegExpReplaceMeta(
       case _ =>
         willNotWorkOnGpu(s"only non-null literal strings are supported on GPU")
     }
+
+    expr.rep match {
+      case Literal(s: UTF8String, DataTypes.StringType) if s != null =>
+        val backrefPattern = Pattern.compile("\\$[0-9]")
```

Comment:
Note that this can create false-positives.  A Spark regex replace string of `"\\$1"` isn't asking for a group but this pattern matcher will think it is.

Reply:
I was working on a fix for this but that exposed [another issue](https://github.com/NVIDIA/spark-rapids/issues/4559) that I am going to have to fix first, so I will mark this PR as a draft for now.